.github/workflows/python-publish.yml

```
@@ -0,0 +1,31 @@
+# This workflows will upload a Python Package using Twine when a release is created
+# For more information see: https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions#publishing-to-package-registries
+
+name: Upload Python Package
+
+on:
+  release:
+    types: [created]
+
+jobs:
+  deploy:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install setuptools wheel twine
+    - name: Build and publish
+      env:
+        TWINE_USERNAME: __token__
```

You forgot to replace this with the pypi username secret.
```suggestion
        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
```

---

No it is correct. Username needs `__token__`

---

I'm confused. Is it a variable of some sort?

---

The template uses the secrets notation. 

https://github.com/actions/starter-workflows/blob/3c3736f59805d1e4f838182263705f44fab9cf68/ci/python-publish.yml#L27

---

No it's a literal.

---

Why is the username `__token__` then? 

Is your password not a password but instead an access token of some sort?

---

That would be correct if using username and password iirc. But the username must be `__token__` when using an api token.

---

> Why is the username `__token__` then? 
> 
> 
> 
> Is your password not a password but instead an access token of some sort?

Yes.

---

Oh I see. I think it's a good idea to leave a comment.

---

```suggestion
        TWINE_USERNAME: __token__ # Using acces-token based login rather than username/password login
```